src/lib/list/selection-list.ts

```diff
@@ -121,6 +127,12 @@ export class MatListOption extends _MatListOptionMixinBase
   /** Whether the label should appear before or after the checkbox. Defaults to 'after' */
   @Input() checkboxPosition: 'before' | 'after' = 'after';
 
+  /** Theme color of the list option. This sets the color of the checkbox. */
+  @Input()
+  get color(): ThemePalette { return this._color || this.selectionList.color; }
```

Comment:
I think that we'll need a `markForCheck` whether the color of the selection list changes, otherwise this won't be picked up.

Reply:
Good point. I think we should add it to the `MatSelectionList#ngOnChanges`, but not sure if we need to add it here. thoughts? We don't have if for other inputs either.

Reply:
Done (added it to the `ngOnChanges`).

Reply:
We have it inside the setter for `disabled`. We could put it in `ngOnChanges` for the theme though.

Reply:
Should we have a test for it as well? I think we have some more `disabled` and `disableRipple` already.

Reply:
Why do we need it though? Isn't `OnPush` supposed to run change detection if an input changes? The case when the input changes for the parent `<mat-selection-list>` should be handled now.

Reply:
The input is changing in a different component though (`MatSelectionList`). `OnPush` only runs change detection for the component whose input was changed. This is the same issue as https://github.com/angular/material2/pull/14858.

Reply:
Yeah exactly, but if `MatSelectionList` changes, then I already mark _all_ options for check. See:

https://github.com/angular/material2/blob/50fa883dd288ae9584eee8a223a77a141e6721cc/src/lib/list/selection-list.ts#L405-L413

Similar to what #14858 does. do I miss something?

Reply:
That change is fine, but I think that we still need a unit test for it so we don't introduce a regression at some point.

Reply:
Cool, it sounded like you wanted me to change that. Probably just some misconception. The current unit test should already cover that.